digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -2071,4 +2132,66 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
       case None => false
     }
   }
-}
\ No newline at end of file
+
+  get("/lanes") {
+    params.get("bbox").map { bbox =>
+      val zoom = params.getOrElse("zoom", halt(BadRequest("Missing zoom"))).toInt
+      val boundingRectangle = constructBoundingRectangle(bbox)
+      val usedService = laneService
+
+      if (zoom >= minVisibleZoom && zoom <= maxZoom) {
+        mapLightLane(usedService.getByZoomLevel( Some(LinkGeomSource.NormalLinkInterface)))
+      } else {
+        validateBoundingBox(boundingRectangle)
+        val assets = usedService.getByBoundingBox(boundingRectangle)
+
+        val updatedInfo = roadAddressService.laneWithRoadAddress(assets)
+        val frozenInfo = roadAddressService.experimentalLaneWithRoadAddress( updatedInfo.map(_.filterNot(_.attributes.contains("VIITE_ROAD_NUMBER"))))
+        mapLanes(updatedInfo ++ frozenInfo)
+      }
+
+    } getOrElse {
+      BadRequest("Missing mandatory 'bbox' parameter")
+    }
+  }
+
+  post("/lanes") {
+    val user = userProvider.getCurrentUser()
+
+    val linkIds = (parsedBody \ "linkIds")extractOrElse[Set[Long]](halt(BadRequest("Malformed 'linkIds' parameter")))
+    val sideCode = (parsedBody \ "sideCode")extractOrElse[Int](halt(BadRequest("Malformed 'sideCode' parameter")))
+    val incomingLanes = (parsedBody \ "lanes").extractOrElse[Seq[NewIncomeLane]](halt(BadRequest("Malformed 'lanes' parameter")))
+
+    validateUserRightsForLanes(linkIds, user)
+    LaneUtils.processNewIncomeLanes(incomingLanes.toSet, linkIds, sideCode, user.username)
+  }
+
+  post("/lanesByRoadAddress") {
+    val user = userProvider.getCurrentUser()
+
+    val sideCode = (parsedBody \ "sideCode")extractOrElse[Int](halt(BadRequest("Malformed 'sideCode' parameter")))
+    val laneRoadAddressInfo = (parsedBody \ "laneRoadAddressInfo").extractOrElse[LaneRoadAddressInfo](halt(BadRequest("Malformed 'laneRoadAddressInfo' parameter")))
+    val incomingLanes = (parsedBody \ "lanes").extractOrElse[Set[NewIncomeLane]](halt(BadRequest("Malformed 'lanes' parameter")))
+
+    validateUserRightsForRoadAddress(laneRoadAddressInfo, user)
+    LaneUtils.processNewLanesByRoadAddress(incomingLanes, laneRoadAddressInfo,sideCode, user.username)
+  }
```

Comment:
This two entry points can be simple transformation in one, but let's apply this only after the deploy to staging if all agree with that:

```scala
  post("/lanes") {
    val user = userProvider.getCurrentUser()

    val linkIds = (parsedBody \ "linkIds").extractOpt[Set[Long]]
    val laneRoadAddressInfo = (parsedBody \ "laneRoadAddressInfo").extractOpt[LaneRoadAddressInfo]
    val sideCode = (parsedBody \ "sideCode")extractOrElse[Int](halt(BadRequest("Malformed 'sideCode' parameter")))
    val incomingLanes = (parsedBody \ "lanes").extractOrElse[Seq[NewIncomeLane]](halt(BadRequest("Malformed 'lanes' parameter")))

    (laneRoadAddressInfo, linkIds) match {
      case (Some(roadAddress), _)=>
        validateUserRightsForRoadAddress(roadAddress, user)
        LaneUtils.processNewLanesByRoadAddress(incomingLanes.toSet, roadAddress,sideCode, user.username)
      case (_, Some(linkids)) =>
        validateUserRightsForLanes(linkids, user)
        LaneUtils.processNewIncomeLanes(incomingLanes.toSet, linkids, sideCode, user.username)
    }
  }

Reply:
Okay for me

Reply:
I can understand the advantages of merging the RoadAddress entry point and lane entry point into one.
But, is it ok to merge 2 things that have different flows into only one?

Reply:
ok, let's keep it like it is.